sdk_extensions/zpages/src/jmh/java/io/opentelemetry/sdk/extensions/zpages/TracezDataAggregatorBenchmark.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.extensions.zpages;
+
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Status;
+import io.opentelemetry.trace.Tracer;
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+
+/** Benchmark class for {@link TracezDataAggregator}. */
+@State(Scope.Benchmark)
+public class TracezDataAggregatorBenchmark {
+
+  private static final String runningSpan = "RUNNING_SPAN";
+  private static final String latencySpan = "LATENCY_SPAN";
+  private static final String errorSpan = "ERROR_SPAN";
+  private final Tracer tracer =
+      OpenTelemetrySdk.getTracerProvider().get("TracezDataAggregatorBenchmark");
+  private final TracezSpanProcessor spanProcessor = TracezSpanProcessor.newBuilder().build();
+  private final TracezDataAggregator dataAggregator = new TracezDataAggregator(spanProcessor);
+
+  @Setup(Level.Trial)
+  public final void setup() {
+    // Generate 1 million running spans, span latencies, and error spans
+    for (int i = 0; i < 1000000; i++) {
```

Comment:
Can you add an `@Param` for this size?

Reply:
Actually I guess we could use a constant for completed spans and a param for running spans, that seems to have more variable overhead. The main observation here is 1 million concurrent requests in a single app is probably not possible for almost all apps!

Reply:
What should the constant be and what values should the param take?